packages/modelence/src/auth/login.ts

```diff
@@ -19,13 +31,39 @@ export async function handleLoginWithPassword(args: Args, { user, session }: Con
     // TODO: handle cases where a user is already logged in
   }
 
-  // TODO: check if the email is verified
   const userDoc = await usersCollection.findOne(
     { 'emails.address': email },
     { collation: { locale: 'en', strength: 2 } }
   );
 
-  const passwordHash = userDoc?.authMethods?.password?.hash;
+  if (!userDoc) {
+    throw incorrectCredentialsError();
+  }
+
+  const emailDoc = userDoc.emails?.find(e => e.address === email);
+
+  if (!emailDoc?.verified && getEmailConfig()?.provider) {
+    if (ip) {
+      try {
+        await consumeRateLimit({
+          bucket: 'verification',
+          type: 'user',
+          value: userDoc._id.toString(),
+        });
+      } catch {
+        throw new Error("Your email address hasn't been verified yet. Please use the verification email we've send earlier to your inbox.");
+      }
+    }
+
+    await sendVerificationEmail({
+      userId: userDoc?._id,
+      email,
+      baseUrl: connectionInfo?.baseUrl,
+    });
+    throw new Error("Your email address hasn't been verified yet. We've sent a new verification email to your inbox.");
+  }
+
+  const passwordHash = userDoc.authMethods?.password?.hash;
```

Comment:
Shouldn't this check be before the email verified checks? If the user is providing a wrong password, we should just reject the login, right? Otherwise you can still brute force unverified emails and trigger the verification emails even though you don't even have the password.